modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/related/models/test/ObjectRelatedModelsProviderTest.java

```diff
@@ -985,6 +991,151 @@ private void _testObjectEntryMtoMObjectRelatedModelsProviderImpl(
 				reverseObjectRelationship.getObjectRelationshipId()));
 	}
 
+	private void _testOrganizationSystemObjectEntry1toMObjectRelatedModelsProviderImpl()
```

Comment:
I saw that this method has a lot of duplicate code with `_testUserSystemObjectEntry1toMObjectRelatedModelsProviderImpl()`, can we make this method more generic and use it for both? this way we can reuse a lot of code.
Let me know if you need any help with this.